tornado/curl_httpclient.py

```diff
@@ -351,6 +351,7 @@ def _curl_setup_request(
             pycurl.HTTPHEADER,
             [
                 "%s: %s" % (native_str(k), native_str(v))
+                if v is not '' else "%s;" % native_str(k)
```

Comment:
This should be `!=` (or just `if v`), not `is not`. The use of `is` here is relying on interning of small strings which I believe is an undocumented implementation detail.